### ISSUE TITLE
Map device buffer for host memory

### ISSUE THREAD

**Issue by LWisteria** [fixstars/clpy#6]
https://github.com/fixstars/clpy/blob/dec25c342032617d61e6b71ad5836b16005cc29a/clpy/core/core.pyx#L2162

**Comment by ghost** [comment]
> We tried to implement an equivalent for `cupy.cuda.PinnedMemory` and found that there isn't correct substitution in OpenCL of allocating pinned host buffer.

https://github.com/fixstars/clpy/pull/144#issuecomment-463496350

The working branch was [6-alloc_host_ptr](https://github.com/fixstars/clpy/tree/6-alloc_host_ptr).

**Comment by LWisteria** [comment]
Yes, I understand. It's very difficult on OpenCL to behaive same as CUDA. Therefore, We decide that ClPy won't support `PinndeMemory` (in the near future at least)

@nsakabe-fixstars can you remove that TODO and leave comment about the problem?
